library/x509.c

```diff
@@ -1019,7 +1019,7 @@ int mbedtls_x509_self_test( int verbose )
         if( verbose != 0 )
             mbedtls_printf( "failed\n" );
 
-        return( ret );
+        goto cleanup;
     }
 
     mbedtls_x509_crt_init( &cacert );
```

Comment:
This initialization needs to happen before the first potential jump to the `cleanup` section, because otherwise we might attempt to call `mbedtls_x509_crt_free()` on an uninitialized CRT structure.